### PR TITLE
chore: deprecate Pin.app

### PR DESCRIPTION
**Pull request by majorgreys**
Deprecate `Pin.app`. Though this property is set in many integrations, it is not set on span data.

### CODE REVIEW

File: ddtrace/pin.py [hunk 1]
```diff
@@ -35,9 +35,10 @@ class Pin(object):
         >>> conn = sqlite.connect('/tmp/image.db')
     """
 
-    __slots__ = ["app", "tags", "tracer", "_target", "_config", "_initialized"]
```
Comment (brettlangdon) [inline ddtrace/pin.py:38]: do we need a removed property for `Pin.app`?

**Review by brettlangdon — COMMENTED** [review]
one question, otherwise lgtm

File: ddtrace/pin.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply ddtrace/pin.py:38]: +1 to this. Looks like `Pin.app` will now be an `AttributeError`, good catch @brettlangdon!

File: ddtrace/pin.py [hunk 2]
```diff
@@ -49,7 +50,7 @@ def __init__(
     ):
         # type: (...) -> None
         tracer = tracer or ddtrace.tracer
-        self.app = app
+        self._app = app
```
Comment (brettlangdon) [inline ddtrace/pin.py:53]: we need `_app` in slots

File: ddtrace/pin.py [hunk 3]
```diff
@@ -35,9 +35,10 @@ class Pin(object):
         >>> conn = sqlite.connect('/tmp/image.db')
     """
 
-    __slots__ = ["app", "tags", "tracer", "_target", "_config", "_initialized"]
+    __slots__ = ["tags", "tracer", "_target", "_config", "_initialized"]
```
Comment (brettlangdon) [inline ddtrace/pin.py:38]:
```suggestion
    __slots__ = ["_app", "tags", "tracer", "_target", "_config", "_initialized"]
```

**Review by brettlangdon — DISMISSED** [review]

**Review by brettlangdon — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

**Pull request merged**